### PR TITLE
CSCFAIRMETA-24: Additional accessibility fixes

### PR DESCRIPTION
**Pull request by vaahtokarkki**
This pull request resolves accessibility issues:

- [x] https://jira.eduuni.fi/browse/CSCFAIRMETA-37  - most of the changes is related to to this issue, all color contrast errors should be fixed (see issue comments for details)
- [x] https://jira.eduuni.fi/browse/CSCFAIRMETA-38 - see issue description for details (Qvain light add dataset page)
- [ ] https://jira.eduuni.fi/browse/CSCFAIRMETA-25 - no changes needed
- [ ] https://jira.eduuni.fi/browse/CSCFAIRMETA-27 - no changes needed
- [ ] https://jira.eduuni.fi/browse/CSCFAIRMETA-28 - no changes needed, Axe errors would need (lots of) refactoring 
- [x] https://jira.eduuni.fi/browse/CSCFAIRMETA-29 - see issue description for details
- [ ] https://jira.eduuni.fi/browse/CSCFAIRMETA-166 - TODO: check is this fixed

Used to test color contrast:
https://addons.mozilla.org/en-US/firefox/addon/wcag-contrast-checker/

### CODE REVIEW

**Comment by vaahtokarkki** [comment]
https://jira.eduuni.fi/browse/CSCFAIRMETA-166 -  Deleted / deprecated banner link text
- Can't find anymore any datasets causing the banner to be visible. **This needs to be tested properly**

**Comment by tahme** [comment]
> https://jira.eduuni.fi/browse/CSCFAIRMETA-166 - Deleted / deprecated banner link text
> 
> * Can't find anymore any datasets causing the banner to be visible. **This needs to be tested properly**
There's a bug that prevents the banner from showing up, this will fix it: https://github.com/CSCfi/etsin-finder/pull/519

File: etsin_finder/frontend/js/components/qvain/datasets/dataset.jsx [hunk 1]
```diff
@@ -189,6 +189,10 @@ const Marker = styled.div`
 `
 
 const DatasetLabel = styled(Label)`
+  color: ${({ props }) => {
+    if (props.color === 'yellow') return props.theme.color.dark
```
Comment (MattiasLevlin) [inline etsin_finder/frontend/js/components/qvain/datasets/dataset.jsx:193]: `props.color` is undefined here and will cause a crash, 

Suggestion: Change row 118 to
`          <Translate color="dark" content="qvain.datasets.oldVersion" component={DatasetLabel} />
`

**Review by MattiasLevlin — CHANGES_REQUESTED** [review]
Checked visual things:

CSCFAIRMETA-37: Vertical scrolling prevented
CSCFAIRMETA-38: Buttons, texts, colors look good
CSCFAIRMETA-29: Font titles added

CSCFAIRMETA-166: Can be tested when:
https://github.com/CSCfi/etsin-finder/pull/519
... has been merged (currently in test)

Overall, looks good! See comment above for requested changes.

File: etsin_finder/frontend/js/components/qvain/datasets/dataset.jsx (hunk 1 above)
Comment (vaahtokarkki) [reply etsin_finder/frontend/js/components/qvain/datasets/dataset.jsx]: Did some small changes and added check that color is defined, defaulting to white. App crash resolved.

**Comment by vaahtokarkki** [comment]
https://jira.eduuni.fi/browse/CSCFAIRMETA-166
Still needs to be checked with the case:
> If there is an older version of the dataset (and it's the only existing version)

Other case:
 
> A new version of this dataset is available. You can open the new version via this link.

Tested and ok

**Review by MattiasLevlin — APPROVED** [review]
https://jira.eduuni.fi/browse/CSCFAIRMETA-166

If there is an older version of the dataset (and it's the only existing version)

... This has now been checked and looks to be functional. Approved.

**Pull request merged**